test/integration/test_large_files.py

```diff
@@ -0,0 +1,195 @@
+######################################################################
+#
+# File: test/integration/test_large_files.py
+#
+# Copyright 2021 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+
+import hashlib
+import io
+import pathlib
+import random
+import string
+import time
+from unittest import mock
+
+import pytest
+
+from b2sdk._v2 import *
+
+from .fixtures import *  # pyflakes: disable
+
+BUCKET_NAME_CHARS = string.ascii_letters + string.digits + '-'
+BUCKET_NAME_LENGTH = 50
+ONE_HOUR_MILLIS = 60 * 60 * 1000
+BUCKET_CREATED_AT_MILLIS = 'created_at_millis'
+
+
+def bucket_name_part(length):
+    return ''.join(random.choice(BUCKET_NAME_CHARS) for _ in range(length))
+
+
+def current_time_millis():
+    """
+    File times are in integer milliseconds, to avoid roundoff errors.
+    """
+    return int(round(time.time() * 1000))
+
+
+def hash_file(path_):
+    hash_ = hashlib.sha1()
+    with open(path_, 'rb') as file_to_hash:
+        while True:
+            data = file_to_hash.read(65536)
+            if not data:
+                break
+            hash_.update(data)
+    return hash_.digest()
```

Comment:
I'm pretty sure we have something like that in utils already

Reply:
almost, I'll add something like that using tools that are there